packages/sanity/src/adapter.ts

```diff
@@ -0,0 +1,156 @@
+import type { Adapter } from "next-auth/adapters"
+import type { Session, Profile, User } from "next-auth"
+import {
+  getUserByIdQuery,
+  getUserByProviderAccountIdQuery,
+  getUserByEmailQuery,
+} from "./queries"
+import LRU from "lru-cache"
+import { SanityClient } from "@sanity/client"
+
+
+const userCache = new LRU<string, User & { id: string }>({
+  maxAge: 24 * 60 * 60 * 1000,
+  max: 1000,
+})
+
+export const SanityAdapter: Adapter<
+  SanityClient,
+  never,
+  User & { id: string },
+  Profile,
+  Session
+> = (client) => {
+  return {
+    async getAdapter() {
+      return {
+        async createUser(profile) {
+          const user = await client.create({
+            _type: "user",
+            email: profile.email,
+            name: profile.name,
+            image: profile.image,
+          })
+
+          userCache.set(user._id, {
+            id: user._id,
+            ...user,
+          })
+
+          return {
+            id: user._id,
+            ...user,
+          }
+        },
+        async getUser(id) {
+          const cachedUser = userCache.get(id)
+
+          if (cachedUser) {
+            ;(async () => {
+              const user = await client.fetch(getUserByIdQuery, {
+                id,
+              })
+
+              userCache.set(user._id, {
+                id: user._id,
+                ...user,
+              })
+            })().then(
+              () => {},
+              () => {}
+            )
```

Comment:
I'm not really sure what is happening in this part, could you try to explain? I also remember that @kripod found out that using `lru-cache` doesn't really help, see: https://github.com/nextauthjs/adapters/issues/70. So maybe we could remove it altogether? What do you think?